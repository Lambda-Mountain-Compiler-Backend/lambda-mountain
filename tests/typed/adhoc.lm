
f := λ(: b Atom). b;
f := λ(: b Nil). b;

f 1;
