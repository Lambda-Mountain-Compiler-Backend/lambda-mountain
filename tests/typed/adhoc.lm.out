(((() (Global (f (: ((App ((App ((App ((Literal :) (Variable b))) (Literal Atom))) (: ((Variable b) Atom)))) (-> (Atom Atom))))))) (Global (f (: ((App ((App ((App ((Literal :) (Variable b))) (Literal Nil))) (: ((Variable b) Nil)))) (-> (Nil Nil))))))) (GExpr (: ((App ((: ((Variable f) (And ((-> (Atom Atom)) (-> (Nil Nil)))))) (: ((Literal 1) Atom)))) Atom))))
